Docs(cloud): added documentation for retries (#967)
MarathonLabs/marathon: docs/cloud/intro/overview.md

When you run tests with Marathon Cloud for the first time, we begin storing the test history in our database. 
The next time you run these tests, we already have information on the average time and the probability of a successful execution for each test. 

> Using this data, we calculate the necessary number of devices to ensure that your tests will complete within 15 minutes, whether that requires 5 devices or 200. 

Reviewer: ```suggestion
Using this data, we calculate the necessary number of devices to ensure that your tests will be completed within 15 minutes, be it 5 or 200.
```

but please note that this may lead to an increase in the number of devices required and the overall time of devices taken for testing. 
However, the total execution time will still be 15 minutes.

> ### Retries
> Retries are a cornerstone of effective testing. The larger your test suite, the greater the likelihood that one test might fail due to environmental instability, commonly known as test flakiness. While adding retries can improve test stability, it also increases the total testing time.
> Marathon offers two retry strategies and several enhancements to strike a balance between stability and efficiency:
> - Predictive retries: By analyzing past test executions, we can determine the success rate of each test. For flaky tests, we can run multiple executions to ensure at least one succeeds.
> - Common retries: If a test fails, we will rerun it up to three times. Marathon will attempt to run the test on a different device to minimize environmental impact on the results.
> - Additional features: Sorting, batching, test rebalancing, and more.
> 

Reviewer: ```suggestion
### Retries
Retries are a cornerstone of effective testing. The larger your test suite, the greater the likelihood that one test might fail due to non-determinism, commonly known as test flakiness. While adding retries can improve test stability, it also increases the total testing time.
Marathon offers two retry strategies and several enhancements to strike a balance between stability and efficiency:
- Predictive retries: By analyzing past test executions, we can determine the success rate of each test. For flaky tests, we can run multiple executions to ensure at least one succeeds.
- Common retries: If a test fails, we will rerun it up to three times. Marathon will attempt to run the test on a different device to minimize the environmental impact on the results.
- Additional features: Sorting, batching, test rebalancing, and more.

```
